packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -231,14 +230,11 @@ class DebugUniversalFramework extends Target {
       .childDirectory('App.framework')
       .childFile('App');
     environment.buildDir.createSync(recursive: true);
-    final RunResult createFrameworkResult = await createStubAppFramework(
+    await _createStubAppFramework(
       output,
-      environment.defines[kSdkRoot],
+      environment,
       iosArchNames,
     );
-    if (createFrameworkResult.exitCode != 0) {
-      throw Exception('Failed to create App.framework.');
-    }
```

Comment:
I don't think this is needed now, the `clang` command will throw if it fails.
https://github.com/flutter/flutter/blob/9f47adc417647abd3c78c3adfa1b2a87d8390313/packages/flutter_tools/lib/src/macos/xcode.dart#L189-L192